doc/reference/networking/net_config.rst

```diff
@@ -1,11 +1,69 @@
 .. _net_config_interface:
 
-Configuration Library
-######################
+Network Configuration Library
+#############################
 
 Overview
 ********
 
+The network configuration library can be used to setup the device in
+semi-automatic way when it is booted. The user just needs to setup
+various Kconfig options and the config library tries to setup the
+device accordingly.
+
+The following Kconfig options affect how configuration library will
+setup the system.
+
+.. csv-table:: Kconfig options for network configuration library
+   :header: "Option name", "Description"
+   :widths: auto
+
+   "CONFIG_NET_CONFIG_SETTINGS", "This option controls whether the network
+   system is configured or initialized at all. If not set, then the config
+   library is not used for initialization and the application needs to do all
+   the network related configuration itself. If this option is set,
+   then optionally user can configure static IP addresses to be set to the
+   first network interface in the system. Typically setting static IP addresses
+   is only usable in testing and should not be used in production code. See
+   the config library Kconfig file for specific options to set the static IP
```

Comment:
Can you provide a reference to this Kconfig file name ( is it ``subsys/net/lib/config/Kconfig``)?

Reply:
yes, that is the file, I will add ref to it